### PR TITLE
userns: further describe minimumKubeletVersion

### PR DESCRIPTION
**Pull request by haircommander**

### CODE REVIEW

File: enhancements/kubelet/user-namespaces-support.md [hunk 1]
```diff
@@ -336,8 +336,44 @@ Thus, we need a method for the kube-apiserver to know for certain a node joining
...
+	// +optional
+	MinimumKubeletVersion string `json:"minimumKubeletVersion,omitempty"`
+}
+```
+
+##### MinmimumKubeletVerison authorization plugin
+
+The operative piece of this feature will be an authorization plugin patched into the kube-apiserver. It will be run against requests coming from
+kubelets, and if kubelet version reported through `Node.Status.NodeSystemInfo.KubeletVersion` is lower than `MinimumKubeletVersion` then the kube-apiserver
+will deny the kubelet from gaining access to any resources that aren't Node get/update and `subjectaccessreviews`. In other words, the kubelet can read it's node
```
Comment (haircommander) [inline enhancements/kubelet/user-namespaces-support.md:361]: WIP implementation https://github.com/openshift/kubernetes/pull/2104

File: enhancements/kubelet/user-namespaces-support.md [hunk 2]
```diff
@@ -336,8 +336,44 @@ Thus, we need a method for the kube-apiserver to know for certain a node joining
...
+type NodeSpec struct {
+	...
+	// MinimumKubeletVersion is the lowest version of a kubelet that can join the cluster
+	// +kubebuilder:validation:Pattern=`^([0-9]*\.[0-9]*\.[0-9]*$`
+	// +optional
+	MinimumKubeletVersion string `json:"minimumKubeletVersion,omitempty"`
+}
+```
+
+##### MinmimumKubeletVerison authorization plugin
```
Comment (wking) [inline enhancements/kubelet/user-namespaces-support.md:357]: nit: `MinmimumKubeletVerison` -> `MinimumKubeletVerison`

Comment (haircommander) [reply enhancements/kubelet/user-namespaces-support.md]: thanks! fixed

File: enhancements/kubelet/user-namespaces-support.md [hunk 3]
```diff
@@ -336,8 +336,44 @@ Thus, we need a method for the kube-apiserver to know for certain a node joining
...
+some pieces to mitigate this pain for cluster admins.
+
+##### MinimumKubeletVersion admission plugin
+
+To protect nodes that are too old from being removed immediately, validation will be run on the MinimumKubeletVersion on admission to see if there are kubelets in the cluster
+that are running with a version lower than the configured version. If so, the creation will be rejected, and the admin notified that they must upgrade their nodes before applying it.
+
+##### MinimumKubeletVersion MCO awareness
+
+MCO will read the MinimumKubeletVersion and mark machines as degraded if the node is not at least MinimumKubeletVersion.
```
Comment (sjenning) [inline enhancements/kubelet/user-namespaces-support.md:374]: > mark machines as degraded

This is fuzzy to me in how the MCO knows the kubelet version on each machine/node, how it is marked as degraded, and what the implications are of a machine being degraded.

**Comment by sjenning** [comment]
Still some fuzzy places for me, but this adds useful information to the existing enhancement

/approve
/lgtm

**Pull request merged**